lms/djangoapps/course_goals/views.py

```diff
@@ -102,3 +104,56 @@ def create(self, post_data):  # lint-amnesty, pylint: disable=arguments-differ
             'is_unsure': goal_key == GOAL_KEY_CHOICES.unsure,
         }
         return JsonResponse(data, content_type="application/json", status=(200 if goal else 201))  # lint-amnesty, pylint: disable=redundant-content-type-for-json-response
+
+
+class PopulateUserActivity(APIView):
+    """
+    API that allows external callers to populate the user activity table
+    """
+    authentication_classes = (JwtAuthentication, SessionAuthentication,)
+    permission_classes = (permissions.IsAuthenticated, IsStaffOrOwner,)
+
+    def post(self, request):
+        """
+        Handle the POST request
+
+        Populate the user activity table.
+        """
+        user_id = request.data.get('user_id')
+        course_key = request.data.get('course_key')
+
+        if not user_id or not course_key:
+            return Response(
+                'User id and course key are required',
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+        try:
+            user_id = int(user_id)
+            user = User.objects.get(id=user_id)
+        except User.DoesNotExist:
+            return Response(
+                'Provided user id does not correspond to an existing user',
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+        try:
+            course_key = CourseKey.from_string(course_key)
+        except InvalidKeyError:
+            return Response(
+                'Provided course key is not valid',
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+        # Populate user activity for tracking progress towards a user's course goals
+        (activity_object_id, created) = UserActivity.populate_user_activity(user, course_key)
+        if activity_object_id:
+            data = {
+                'activity_object_id': activity_object_id
```

Comment:
isn't returning the id of a created object somewhat standard?  
i think this would also be helpful for testing their implementation/debugging any issues that could come up
i guess if you think this is an anti-pattern of sorts i could just return a text response?

Reply:
I do think it's an anti-pattern if you aren't building a RESTful API - and I don't think that's what you're doing here. The client never does anything with this ID value, right? There's no CRUD style "delete this user activity record" or "update this record" calls.

You don't need a text response do you? Just a 200 code?

Reply:
sure, i guess i can just return the code and i'll add a log so mobile team can debug